src/proto/h2/client.rs

```diff
@@ -305,7 +302,7 @@ pin_project! {
         T: Unpin,
     {
         #[pin]
-        drop_rx: StreamFuture<Receiver<Infallible>>,
+        drop_rx: Receiver<Infallible>,
```

Comment:
I kept the changes here to a minimum to minimize the risk of regressions, but I wonder if this couldn't be a plain old `AtomicBool`?

Reply:
I guess it depends if the bug mention above is handled. I think this was a oneshot channel so that the task would be woken up again.